src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -13,23 +13,94 @@
+//   4) str.Equals(String, Ordinal or OrdinalIgnoreCase)
+//   5) str.StartsWith(string, Ordinal or OrdinalIgnoreCase)
 //   6) MemoryExtensions.SequenceEqual<char>(ROS<char>, ROS<char>)
-//   7) MemoryExtensions.Equals(ROS<char>, ROS<char>, StringComparison.Ordinal)
+//   7) MemoryExtensions.Equals(ROS<char>, ROS<char>, Ordinal or OrdinalIgnoreCase)
 //   8) MemoryExtensions.StartsWith<char>(ROS<char>, ROS<char>)
-//   9) MemoryExtensions.StartsWith(ROS<char>, ROS<char>, StringComparison.Ordinal)
+//   9) MemoryExtensions.StartsWith(ROS<char>, ROS<char>, Ordinal or OrdinalIgnoreCase)
 //
 // When one of the arguments is a constant string of a [0..32] size so we can inline
 // a vectorized comparison against it using SWAR or SIMD techniques (e.g. via two V256 vectors)
 //
-// We might add these in future:
-//   1) OrdinalIgnoreCase for everything above
-//   2) Span.CopyTo
-//   3) Spans/Arrays of bytes (e.g. UTF8) against a constant RVA data
+
+//------------------------------------------------------------------------
+// ConvertToLowerCase: Converts input ASCII data to lower case
+//
+// Arguments:
+//    input  - Constant data to change casing to lower
+//    mask   - Mask to apply to non-constant data, e.g.:
+//       input: [  h ][  i ][  4 ][  - ][  A ]
+//       mask:  [0x20][0x20][ 0x0][ 0x0][0x20]
+//    length - Length of input
+//
+// Return Value:
+//    false if input contains non-ASCII chars
 //
+static bool ConvertToLowerCase(WCHAR* input, WCHAR* mask, int length)
+{
+    for (int i = 0; i < length; i++)
+    {
+        auto ch = (USHORT)input[i];
+        if (ch > 127)
+        {
+            JITDUMP("Constant data contains non-ASCII char(s), give up.\n");
+            return false;
+        }
+
+        // Inside [0..127] range only [a-z] and [A-Z] sub-ranges are
+        // eligible for case changing, we can't apply 0x20 bit for e.g. '-'
+        if (((ch >= 'A') && (ch <= 'Z')) || ((ch >= 'a') && (ch <= 'z')))
+        {
+            input[i] |= 0x20;
+            mask[i] = 0x20;
+        }
+        else
+        {
+            mask[i] = 0;
+        }
+    }
+    return true;
+}
+
+#if defined(FEATURE_HW_INTRINSICS) && defined(TARGET_64BIT)
+//------------------------------------------------------------------------
+// CreateConstVector: a helper to create Vector128/256.Create(<cns>) node
+//
+// Arguments:
+//    comp     - Compiler object
+//    simdType - Vector type, either TYP_SIMD32 (xarch only) or TYP_SIMD16
+//    cns      - Constant data
+//
+// Return Value:
+//    GenTreeHWIntrinsic node representing Vector128/256.Create(<cns>)
+//
+static GenTreeHWIntrinsic* CreateConstVector(Compiler* comp, var_types simdType, WCHAR* cns)
+{
+    const CorInfoType baseType = CORINFO_TYPE_ULONG;
+
+    // We can use e.g. UINT here to support SIMD for 32bit as well,
+    // but it significantly complicates code, so 32bit support is left up-for-grabs
+    assert(sizeof(ssize_t) == 8);
```

Comment:
You could use `static_assert(...)` here, so the code would never be enabled.

Reply:
good idea, will change in a follow up PR